app/workers/functions.php

```diff
@@ -259,8 +259,16 @@ private function execute(
         $execution->setAttribute('status', 'processing');
         $execution = $dbForProject->updateDocument('executions', $executionId, $execution);
 
+        $vars = [];
+
+        $variables = $function['vars'];
+
+        foreach ($variables as $variable) {
+            $vars[$variable['key']] = $variable['value'];
+        }
+
```

Comment:
we should do the same as my suggestion in the controller to make this more safe

Reply:
Updated on multiple places.